cluster-autoscaler/cloudprovider/aws/aws_wrapper.go

```diff
@@ -107,59 +109,50 @@ func (m *awsWrapper) getAutoscalingGroupsByNames(names []string) ([]*autoscaling
 	return asgs, nil
 }
 
-func (m *awsWrapper) getAutoscalingGroupNamesByTags(kvs map[string]string) ([]string, error) {
-	// DescribeTags does an OR query when multiple filters on different tags are
-	// specified. In other words, DescribeTags returns [asg1, asg1] for keys
-	// [t1, t2] when there's only one asg tagged both t1 and t2.
+func (m *awsWrapper) getAutoScalingGroupsByTags(configs []asgAutoDiscoveryConfig) ([]*autoscaling.Group, error) {
+	if len(configs) == 0 {
+		return nil, nil
+	}
+
 	filters := []*autoscaling.Filter{}
-	for key, value := range kvs {
-		filter := &autoscaling.Filter{
-			Name:   aws.String("key"),
-			Values: []*string{aws.String(key)},
-		}
-		filters = append(filters, filter)
-		if value != "" {
-			filters = append(filters, &autoscaling.Filter{
-				Name:   aws.String("value"),
-				Values: []*string{aws.String(value)},
-			})
+	tagNames := []string{}
+	for _, config := range configs {
+		for key, value := range config.Tags {
+			filter := &autoscaling.Filter{
+				Name:   aws.String("tag-key"),
+				Values: []*string{aws.String(key)},
+			}
+			filters = append(filters, filter)
+			if value != "" {
+				filters = append(filters, &autoscaling.Filter{
+					Name:   aws.String("tag-value"),
+					Values: []*string{aws.String(value)},
+				})
+			}
+			tagNames = append(tagNames, key)
 		}
 	}
+	klog.V(4).Infof("Regenerating instance-to-ASG map for auto-discovery tags: %v", tagNames)
 
-	tags := []*autoscaling.TagDescription{}
-	input := &autoscaling.DescribeTagsInput{
+	asgs := make([]*autoscaling.Group, 0)
```

Comment:
just curious, is there a difference between using `asgs := make([]*autoscaling.Group, 0)` vs `asgs := []*autoscaling.Group{}` ? I can't remember

Reply:
They're functionally the same, in this case each creating a slice of length 0. I don't have any strong preference between the syntaxes, so if one is more idiomatic, I can change it.

Reply:
I'd change to `[]*autoscalingGroup{}`